File: Models/Project.cs


using System.ComponentModel.DataAnnotations;

namespace DataPlane.Models
{
    public class Project
    { 

        // Primary key
        public int ProjectId { get; set; }

        [Required]
        public string Name { get; set; }
        public string Description { get; set; }

        // Navigation properties 
        public virtual ProjectType ProjectType { get; set; }
    }
}

File: ClientApp/src/app/models/project.ts
import { Basemodel } from './basemodel';

export interface Project {

    projectId: number;

    name: string;
    description: string;
    projectTypeId: number;

}

export class Project extends Basemodel {

    public projectId: number;

    public name: string;
    public description: string;
    public projectTypeId: number;

}
